src/main/java/twilight/Parser.java

```diff
@@ -0,0 +1,53 @@
+package twilight;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.lang.IndexOutOfBoundsException;
+
+/**
+ * Represents a class used for transforming user input to a command to be carried out by chatbot.
+ */
+public class Parser {
+    private static final String[] COMMANDS = new String[]{"list", "mark", "unmark", "todo", "event", "deadline", "delete", "find", "bye"};
+    public Parser(){}
+
+    /**
+     * Parses the string input and returns a command to be executed.
+     *
+     * @param input The message from user.
+     * @return Command according to user desire.
+     * @throws InvalidInputException When the message from user is an invalid statement.
+     */
+    public static Command parse(String input) throws InvalidInputException {
+        try {
+            String[] split = input.split(" ", 2);
+            String command = split[0];
+            String details = split[1];
+            int i = 0;
+            while (!COMMANDS[i].equals(command)) {
+                i++;
+            }
+            if (i == 0) {
```

Comment:
Would using a switch statement be cleaner here? Also, perhaps checking if the command equate to the exact string command (i.e. `command == "event"`) would make your code more readable?